File: NETWORKS_LAB/A5/SOURCE_CODE/checker.cpp
#include <iostream>
#include <fstream>
#include <string>

int main() {
    std::ifstream input("input.txt");
    std::ifstream output("output.txt");

    if (!input.is_open() || !output.is_open()) {
        std::cerr << "Error opening files!" << std::endl;
        return 1;
    }

    std::string input_line, output_line;
    int line_number = 0;
    bool mismatch_found = false;

    while (std::getline(input, input_line) && std::getline(output, output_line)) {
        line_number++;

        if (input_line != output_line) {
            mismatch_found = true;
            std::cout << "Mismatch found at line " << line_number << std::endl;
            std::cout << "Input:  " << input_line << std::endl;
            std::cout << "Output: " << output_line << std::endl;

            // Print 100 characters after the first mismatch
            std::string::size_type pos = input_line.find_first_not_of(output_line);
            if (pos != std::string::npos) {
                std::cout << "100 characters after mismatch:" << std::endl;
                std::cout << input_line.substr(pos, 100) << std::endl;
            }
            break;
        }
    }

    if (!mismatch_found) {
        std::cout << "Files are identical." << std::endl;
    }

    input.close();
    output.close();

    return 0;
}

File: NETWORKS_LAB/A5/SOURCE_CODE/checker.py
def main():
    with open('input.txt', 'r') as input_file, open('output.txt', 'r') as output_file:
        input_lines = input_file.readlines()
        output_lines = output_file.readlines()

        mismatch_found = False

        for i, (input_line, output_line) in enumerate(zip(input_lines, output_lines)):
            if input_line != output_line:
                mismatch_found = True
                print(f"Mismatch found at line {i + 1}:")
                print(f"Input:  {input_line.strip()}")
                print(f"Output: {output_line.strip()}")

                # Print 100 characters after the first mismatch
                pos = next((pos for pos, (a, b) in enumerate(zip(input_line, output_line)) if a != b), None)
                if pos is not None:
                    print("100 characters after mismatch:")
                    print(input_line[pos:pos+100])
                break

        if not mismatch_found:
            print("Files are identical.")

if __name__ == "__main__":
    main()
